src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -330,16 +330,19 @@ class RedownloadService[F[_]: NonEmptyParallel](
     def wrappedRedownload(
       shouldRedownload: Boolean,
       meaningfulAcceptedSnapshots: SnapshotsAtHeight,
-      meaningfulMajorityState: SnapshotsAtHeight
+      meaningfulMajorityState: SnapshotsAtHeight,
+      maxMajorityHeight: Long
     ) =
       for {
         _ <- if (shouldRedownload) {
-          val plan = calculateRedownloadPlan(meaningfulAcceptedSnapshots, meaningfulMajorityState)
+          val redownloadPlan = calculateRedownloadPlan(meaningfulAcceptedSnapshots, meaningfulMajorityState)
+          val downloadPlan = calculateDownloadPlan(redownloadPlan, maxMajorityHeight)
```

Comment:
@marcinwadon I'm wondering, do we have all the blocks we need in the highest snapshot? I remember there was a acceptedCBSinceSnapshotCache and checkpointCache in StoredSnapshot but I wonder if we will not trigger resolving because of missing parents?

Reply:
Everything which is not in the snapshot, should be in snapshot info. Hence, downloading both snapshot and snapshot info should download everything what is needed.